tests/MiddlewareTest.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Spatie\Permission\Test;
+
+use Auth;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Spatie\Permission\Middleware\RoleMiddleware;
+use Spatie\Permission\Middleware\PermissionMiddleware;
+
+class MiddlewareTest extends TestCase
+{
+    /**
+     * @test
+     * @expectedException Symfony\Component\HttpKernel\Exception\HttpException
+     */
+    public function it_can_not_access_with_middleware_role_if_is_guest()
```

Comment:
The name of this test feels a bit off. Could you make this more readable eg `a_guest_cannot_access_a_route_protected_by_the_role_middleware` or something better?